ui/app/components/two-step-button.js

```diff
@@ -33,6 +34,9 @@ export default class TwoStepButton extends Component {
     while (true) {
       let ev = yield waitForEvent(document.body, 'click');
       if (!this.element.contains(ev.target) && !this.awaitingConfirmation) {
+        if (this.onCancel) {
+          this.onCancel();
+        }
```

Comment:
Surprised this is needed here! onCancel() isn't a new convention for two-step buttons. Has it just never been used in practice before this?